pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -6571,30 +6597,45 @@ spec:
 	prt := newPipelineRunTest(t, d)
 	defer prt.Cancel()
 
-	run1, _ := prt.reconcileRun("foo", "pipelinerun-param-invalid-result-variable", nil, true)
-	run2, _ := prt.reconcileRun("foo", "pipelinerun-pipeline-result-invalid-result-variable", nil, true)
-	run3, _ := prt.reconcileRun("foo", "pipelinerun-with-optional-workspace-validation", nil, true)
-
-	cond1 := run1.Status.GetCondition(apis.ConditionSucceeded)
-	cond2 := run2.Status.GetCondition(apis.ConditionSucceeded)
-	cond3 := run3.Status.GetCondition(apis.ConditionSucceeded)
-
-	for _, c := range []*apis.Condition{cond1, cond2, cond3} {
-		if c.Status != corev1.ConditionFalse {
-			t.Errorf("expected Succeeded/False condition but saw: %v", c)
-		}
-	}
-
-	if cond1.Reason != ReasonInvalidTaskResultReference {
-		t.Errorf("expected invalid task reference condition but saw: %v", cond1)
-	}
-
-	if cond2.Reason != ReasonInvalidTaskResultReference {
-		t.Errorf("expected invalid task reference condition but saw: %v", cond2)
+	testCases := []struct {
+		name      string
+		wantErr   string
+		condition apis.ConditionType
```

Comment:
is a separate variable needed for condition succeeded?

Reply:
Hardcoded